features/steps/helpers.py

```diff
@@ -1,20 +1,49 @@
+from time import sleep
+
+from selenium.common.exceptions import ElementClickInterceptedException
+from selenium.common.exceptions import ElementNotInteractableException
 from selenium.webdriver.common.by import By
 from selenium.webdriver.support import expected_conditions
 from selenium.webdriver.support.wait import WebDriverWait
 
 
 def sign_in(context, login=None, browser=None):
     if login is None:
-        login = context.user.email
+        login = context.username
     if browser is None:
         browser = context.browser
     browser.get(context.base_url)
     login_form = WebDriverWait(browser, 10).until(
         expected_conditions.visibility_of_element_located((By.CLASS_NAME, "login"))
     )
-    username = login_form.find_element_by_xpath(".//input[@id='username']")
+    cautious_send_keys(browser=login_form, xpath=".//input[@id='username']", sequence=login)
     password = login_form.find_element_by_xpath(".//input[@id='password']")
-    username.send_keys(login)
     password.send_keys("userpasswd")
     submit_button = login_form.find_element_by_xpath(".//input[@type='submit']")
     submit_button.click()
+
+
+def cautious_click(browser, xpath, iterations=10, timeout=1):
+    iterations = iterations - 1
+    element = browser.find_element_by_xpath(xpath)
+    try:
+        sleep(timeout)
+        element.click()
+    except ElementClickInterceptedException as Error:
+        if iterations > 0:
+            cautious_click(browser, xpath, iterations, timeout)
+        else:
+            raise Error
+
+
+def cautious_send_keys(browser, xpath, sequence, iterations=10, timeout=1):
```

Comment:
it is almost identical to `cautious_click`. Please generalize it to `cautious_action` and add callable argument `action` that is to be performed on `element`

Reply:
Done